docs/source/user_guide_kg_builder.rst

```diff
@@ -0,0 +1,410 @@
+
+    To use OpenAI (embedding or LLM), the `OPENAI_API_KEY` must be in the env vars, for instance using:
+
+    .. code:: python
+
+        import os
+        os.environ["OPENAI_API_KEY"] = "sk-..."
+
+
+If OpenAI is not an option, see :ref:`embedders` to learn how to use sentence-transformers or create your own embedder.
+
+The embeddings are added to each chunk metadata, and will be saved as a Chunk node property in the graph if
+`create_lexical_graph` is enabled in the `EntityRelationExtractor` (keep reading).
+
+
+Schema Builder
+==============
+
+The schema is used to try and ground the LLM to a list of possible entities and relations of interest.
+So far, schema must be manually created by specifying:
+
+- **Entities** the LLM should look for in the text, including their properties (name and type).
+- **Relations** of interest between these entities, including the relation properties (name and type).
+- **Triplets** to define the start (source) and end (target) entity types for each relation.
+
+Here is a code block illustrating these concepts:
+
+.. code:: python
+
+    from neo4j_genai.experimental.components.schema import (
+        SchemaBuilder,
+        SchemaEntity,
+        SchemaProperty,
+        SchemaRelation,
+    )
+
+    schema_builder = SchemaBuilder()
+
+    schema_builder.run(
+        entities=[
+            SchemaEntity(
+                label="Person",
+                properties=[
+                    SchemaProperty(name="name", type="STRING"),
+                    SchemaProperty(name="place_of_birth", type="STRING"),
+                    SchemaProperty(name="date_of_birth", type="DATE"),
+                ],
+            ),
+            SchemaEntity(
+                label="Organization",
+                properties=[
+                    SchemaProperty(name="name", type="STRING"),
+                    SchemaProperty(name="country", type="STRING"),
+                ],
+            ),
+        ],
+        relations=[
+            SchemaRelation(
+                label="WORKED_ON",
+            ),
+            SchemaRelation(
+                label="WORKED_FOR",
+            ),
+        ],
+        possible_schema=[
+            ("Person", "WORKED_ON", "Field"),
+            ("Person", "WORKED_FOR", "Organization"),
+        ],
+    )
+
+After validation, this schema is saved in a `SchemaConfig` object, whose dict representation is passed
+to the LLM.
+
+
+Entity and Relation Extractor
+=============================
+
+This component is responsible for extracting the relevant entities and relationships from each text chunk,
+using the schema as guideline.
+
+This package contains an LLM-based entity and relationships extractor: `LLMEntityRelationExtractor`.
+It can be used in this way:
+
+.. code:: python
+
+    from neo4j_genai.experimental.components.entity_relation_extractor import (
+        LLMEntityRelationExtractor,
+    )
+    from neo4j_genai.llm import OpenAILLM
+
+    extractor = LLMEntityRelationExtractor(
```

Comment:
I would add a section on inserting a custom prompt / prompt template. I think most users will be very interested in that

Reply:
Isn't this covered here: https://github.com/neo4j/neo4j-genai-python/pull/102/files#diff-c37850688512e77f403fe5c0f70c11f569a5ae233777f9859877f2f49dfb4619R306 ?

Reply:
Sorry I missed that!

Reply:
No pb, thanks for the review!